src/test/java/com/amazon/opendistroforelasticsearch/security/protected_indices/ProtectedIndicesTests.java

```diff
@@ -0,0 +1,578 @@
+    }
+
+    /************************************************************************************************
+     * Tests with a user who has all index permissions but is not member of the protectedIndexRoles
+     ***********************************************************************************************/
+
+    // Test data search
+    @Test
+    public void testNoSearchResults() throws Exception {
+        createTestIndicesAndDocs();
+
+        // Create rest client
+        RestHelper rh = nonSslRestHelper();
+
+        // Test direct index query.
+        for (String index : listOfIndexesToTest) {
+            RestHelper.HttpResponse response = rh.executePostRequest(index + "/_search", matchAllQuery, indexAccessNoRoleUserHeader);
+            XContentParser xcp = XContentType.JSON.xContent().createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, response.getBody());
+            SearchResponse searchResponse = SearchResponse.fromXContent(xcp);
+            // confirm good response.
+            assertTrue(searchResponse.status() == RestStatus.OK);
+            // confirm no search hits.
+            assertTrue(searchResponse.getHits().getHits().length == 0);
+            // confirm no failed shards.
+            assertTrue(searchResponse.getFailedShards() == 0);
+            // confirm data was actually queried
+            assertTrue(searchResponse.getSuccessfulShards() == 5);
+        }
+
+        // Test index pattern
+        for (String indexPattern : listOfIndexPatternsToTest) {
+            RestHelper.HttpResponse response = rh.executePostRequest(indexPattern + "/_search", matchAllQuery, indexAccessNoRoleUserHeader);
+            XContentParser xcp = XContentType.JSON.xContent().createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, response.getBody());
+            SearchResponse searchResponse = SearchResponse.fromXContent(xcp);
+            // confirm good response.
+            assertTrue(searchResponse.status() == RestStatus.OK);
+            // confirm no search hits.
+            assertTrue(searchResponse.getHits().getHits().length == 0);
+            // confirm no failed shards.
+            assertTrue(searchResponse.getFailedShards() == 0);
+            // confirm data was actually queried
+            assertTrue(searchResponse.getSuccessfulShards() == 15);
+        }
+    }
+
+    @Test
+    public void testNoResultsAlias() throws Exception {
+        createTestIndicesAndDocs();
+
+        int i = 0;
+        try (TransportClient tc = getInternalTransportClient()) {
+            for (String index : listOfIndexesToTest) {
+                IndicesAliasesRequest request = new IndicesAliasesRequest();
+                IndicesAliasesRequest.AliasActions aliasAction =
+                        new IndicesAliasesRequest.AliasActions(IndicesAliasesRequest.AliasActions.Type.ADD)
+                                .index(index)
+                                .alias("alias" + i);
```

Comment:
Shouldn't the ```i``` be incremented in this loop ?

Reply:
good catch.